test/test_data/test_target_validator.py

```diff
@@ -176,66 +156,87 @@ def test_targetvalidator_supported_types_noclassification(input_data_targettest)
-            assert np.max(transformed_y) == len(np.unique(transformed_y)) - 1
+    # Assert dtype is presevered with y -> y_encoded -> y_inverse
+    def dtype(arr):
+        if isinstance(arr, list):
+            if isinstance(arr[0], list):
+                return type(arr[0][0])
+            else:
+                return type(arr[0])
+        elif isinstance(arr, pd.DataFrame):
+            return arr.dtypes[0]
         else:
-            for col in range(transformed_y.shape[1]):
-                assert np.min(transformed_y[:, col]) == 0
-                assert np.max(transformed_y[:, col]) == len(np.unique(transformed_y[:, col])) - 1
-
-        # Make sure we can perform inverse transform
-        y_inverse = validator.inverse_transform(transformed_y)
-        if hasattr(input_data_targettest, 'dtype'):
-            # In case of numeric, we need to make sure dtype is preserved
-            if is_numeric_dtype(input_data_targettest.dtype):
-                assert y_inverse.dtype == input_data_targettest.dtype
-            # Then make sure every value is properly inverse-transformed
-            np.testing.assert_array_equal(np.array(y_inverse), np.array(input_data_targettest))
-        elif hasattr(input_data_targettest, 'dtypes'):
-            if is_numeric_dtype(input_data_targettest.dtypes[0]):
-                assert y_inverse.dtype == input_data_targettest.dtypes[0]
-            # Then make sure every value is properly inverse-transformed
-            np.testing.assert_array_equal(np.array(y_inverse),
-                                          # pandas is always (N, 1) but targets are ravel()
-                                          input_data_targettest.to_numpy().reshape(-1))
+            return arr.dtype
+
+    if is_numeric_dtype(dtype(y)):
+        assert y_inverse.dtype == dtype(y)
+
+    # Assert that y == y_inverse, giving back the original values
+    np.testing.assert_array_equal(y_inverse, np.array(y).reshape(-1))
+
+    # Assert the y_inverse, y_encoded have the expected shape
+    shape = np.shape(y)
+    if len(shape) == 2 and shape[1] == 1:
+        # For cases where y = [[1], [2], [3]],
+        # we expect y_inverse, y_encodedd to have been flattened to [1,2,3]
+        expected_shape = (shape[0], )
     else:
-        # Sparse is not encoded, mainly because the sparse data is expected
-        # to be numpy of numerical type -- which currently does not require encoding
-        np.testing.assert_array_equal(
-            np.ravel(input_data_targettest.todense()),
-            np.ravel(transformed_y.todense())
-        )
+        expected_shape = shape
+
+    assert y_encoded.shape == expected_shape
+    assert y_inverse.shape == expected_shape
+
+    # These next part of the tests rely on some encoding to have taken place
+    # This happens when `is_classification` and not task_type = multilabel-indicator
+    #
+    # TargetValidator._fit()
+    # > Also, encoding multilabel indicator data makes the data multiclass
```

Comment:
Does this happen in practice?

Reply:
I have no idea, this was just a comment I had found in `TargetValidator._fit()` which was relevant to the test so I included it here as well.

Reply:
Got it, so we're not doing anything here. Maybe you could update the comment to reflect this better?

Reply:
Update comment